x/concentrated-liquidity/incentives.go

```diff
@@ -1004,40 +973,44 @@ func (k Keeper) GetClaimableIncentives(ctx sdk.Context, positionId uint64) (sdk.
 // Returns error if:
 // - position with the given id does not exist
 // - other internal database or math errors.
-func (k Keeper) collectIncentives(ctx sdk.Context, sender sdk.AccAddress, positionId uint64) (sdk.Coins, sdk.DecCoins, error) {
+func (k Keeper) collectIncentives(ctx sdk.Context, sender sdk.AccAddress, positionId uint64) (sdk.Coins, sdk.Coins, error) {
 	// Retrieve the position with the given ID.
 	position, err := k.GetPosition(ctx, positionId)
 	if err != nil {
-		return sdk.Coins{}, sdk.DecCoins{}, err
+		return sdk.Coins{}, sdk.Coins{}, err
 	}
 
 	err = k.ensurePositionOwner(ctx, sender, position.PoolId, positionId)
 	if err != nil {
-		return sdk.Coins{}, sdk.DecCoins{}, err
+		return sdk.Coins{}, sdk.Coins{}, err
 	}
 
 	// Claim all incentives for the position.
-	collectedIncentivesForPosition, forfeitedIncentivesForPosition, err := k.claimAllIncentivesForPosition(ctx, position.PositionId)
+	collectedIncentivesForPosition, forfeitedIncentivesForPosition, err := k.prepareClaimAllIncentivesForPosition(ctx, position.PositionId)
 	if err != nil {
-		return sdk.Coins{}, sdk.DecCoins{}, err
+		return sdk.Coins{}, sdk.Coins{}, err
 	}
 
 	// If no incentives were collected, return an empty coin set.
-	// TODO: Does collectedIncentivesForPosition = 0 => forfeitedIncentivesForPosition = 0?
-	// TODO: Do we need to get event emitted here still?
-	if collectedIncentivesForPosition.IsZero() {
+	if collectedIncentivesForPosition.IsZero() && forfeitedIncentivesForPosition.IsZero() {
```

Comment:
Do we have tests for:
- only `collectedIncentivesForPosition` is zero
- only `forfeitedIncentivesForPosition` is zero
?

Reply:
Yes! Both these (and other combinations) are tested in `TestQueryAndCollectIncentives`